### ISSUE TITLE
eval crashes the browser

### ISSUE THREAD

**Issue by pycage** [sailfishos/sailfish-browser#676]
While working a lot with loading JavaScript modules dynamically, I noticed that the browser was prone to crash.
I was able to reduce the crash scenario to this minimal example, which almost certainly crashes
the browser every time:

```
<!DOCTYPE html>
<html>
<body>
    <script>
        for (let i = 0; i < 100000; ++i) eval("const a = 42;");
    </script>
</body>
</html>
```

**Comment by rainemak** [comment]
Thank you for the report. We'll give a look.

**Comment by llewelld** [comment]
Here's a backtrace for the crash, running the [nemo/45.9.1+git19](https://git.sailfishos.org/mer-core/gecko-dev/-/tags/nemo%2F45.9.1+git19
) version of gecko-dev:

[backtrace.txt](https://github.com/sailfishos/sailfish-browser/files/4658453/backtrace.txt)

It looks like this crash may have been fixed by @adenexter's gecko-dev commit [72e2b943](https://git.sailfishos.org/mer-core/gecko-dev/commit/72e2b943eea2c8f44f35113d4738fea66309c858) (not in the current release build). I'm just double checking this now.

**Comment by llewelld** [comment]
Apparently it's not quite as simple as just applying [72e2b943](https://git.sailfishos.org/mer-core/gecko-dev/commit/72e2b943eea2c8f44f35113d4738fea66309c858). However, applying that along with the [9a33fc33](https://git.sailfishos.org/mer-core/gecko-dev/commit/9a33fc333480346eec2f51097456345bbf59cb97) does seem to fix the issue.

**Comment by llewelld** [comment]
I take back my previous comment. Commit 72e2b943 is indeed enough to address this. The following PR is enough to fix it for the next release: https://git.sailfishos.org/mer-core/gecko-dev/merge_requests/96

@pycage: thanks for your excellent report and finding a minimal example for triggering the bug. It's been a big help.

**Comment by llewelld** [comment]
The changes for this have all been merged into gecko-dev master and the upgrade-3.3.0 branch, and will make it into the next release, so I'll close this now.

**Comment by rainemak** [comment]
Thank you @llewelld :+1:
